Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -8643,6 +8643,17 @@ void HTMLMediaElement::purgeBufferedDataIfPossible()
 {
     ALWAYS_LOG(LOGIDENTIFIER);
 
+    bool isPausedAndNoMSE = [&] {
+#if ENABLE(MEDIA_SOURCE)
+        if (m_mediaSource)
+            return false;
+#endif
+        return paused();
+    }();
+
+    if (isPausedAndNoMSE)
```

Comment:
s/isPausedAndNoMSE/isPausedAndNoMSE()/

Reply:
done.

Reply:
Why was the test passing? Should we add a test that checks the false case?

Reply:
why would it not pass?. The lambda was evaluated before and the result set to the value.

Reply:
actually, on second thought, I like the original way better, and avoid to have to keep the lambda for longer than necessary

Reply:
> Should we add a test that checks the false case?
in the non MSE case, how the data is evicted (and if it's going to be) is managed by AVFoundation I'm not sure how this could be tested in a guaranteed outcome

Reply:
Oh I misread the code, sorry about the noise. Yes this is correct as is.